infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/RoutingQueuesSnapshotReader.java

```diff
@@ -244,55 +240,56 @@ private void waitForData() {
         }
     }
 
-    // Check if the Opaque Entry contains an End Marker, denoting the end of snapshot sync data
-    private boolean isEndMarker(OpaqueEntry opaqueEntry) {
-        if (!opaqueEntry.getEntries().keySet().contains(endMarkerStreamId)) {
-            return false;
-        }
-        log.info("isEndMarker is called from next of snapshot reader");
-        // There should be a single SMR update for the end marker
-        Preconditions.checkState(opaqueEntry.getEntries().get(endMarkerStreamId).size() == 1);
-        SMREntry smrEntry = opaqueEntry.getEntries().get(endMarkerStreamId).get(0);
+    private CorfuStoreEntry<Queue.RoutingQSnapSyncHeaderKeyMsg, Queue.RoutingQSnapSyncHeaderMsg, Message>
+    getFullSyncTxnHeader(OpaqueEntry opaqueEntry) {
+        Preconditions.checkState(opaqueEntry.getEntries().get(snapSyncHeaderStreamId).size() == 1);
```

Comment:
the marker is written in the first transaction only, wont this precondition get violated for subsequent updates?